src/overlays/actors/ovl_Bg_Tobira01/z_bg_tobira01.c

```diff
@@ -28,14 +28,74 @@ const ActorInit Bg_Tobira01_InitVars = {
     (ActorFunc)BgTobira01_Update,
     (ActorFunc)BgTobira01_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Tobira01/func_80B12430.s")
+void BgTobira01_Action(BgTobira01* this, PlayState* play) {
+    Player* sp1C = GET_PLAYER(play);
+    s16 actorCutscene = this->dyna.actor.csId;
+    s16 tempDoorYPosition;
+    if (this->cutscenePlayed != 0) {
+        if (CutsceneManager_GetCurrentCsId() == CS_ID_GLOBAL_TALK) {
+            CutsceneManager_Stop(CS_ID_GLOBAL_TALK);
+        } else if (CutsceneManager_IsNext(actorCutscene) != 0) {
+            CutsceneManager_StartWithPlayerCs(actorCutscene, &this->dyna.actor);
+            gSaveContext.save.saveInfo.weekEventReg[88] |= 0x40;
+            this->cutscenePlayed = 0;
+        } else {
+            CutsceneManager_Queue(actorCutscene);
+        }
+    } else if (!(gSaveContext.save.saveInfo.weekEventReg[88] & 0x40) && (this->doorYPositionTick == 0) &&
+               (play->actorCtx.playerImpact.timer != 0) && (play->actorCtx.playerImpact.type == 0) &&
+               (SurfaceType_GetSceneExitIndex(&play->colCtx, sp1C->actor.floorPoly, sp1C->actor.floorBgId) == 6)) {
+        this->cutscenePlayed = 1;
+        this->unused = 0;
+    }
+
+    tempDoorYPosition = this->doorYPositionTick;
+    if (gSaveContext.save.saveInfo.weekEventReg[88] & 0x40) {
+        this->doorYPositionTick += 1;
+    } else {
+        this->doorYPositionTick -= 1;
+    }
+
+    this->doorYPositionTick = CLAMP(this->doorYPositionTick, 0, 60);
```

Comment:
```suggestion
    this->doorYPositionTick = CLAMP(this->doorYPositionTick, 0, 60);
```
Could define 0 and 60 as DOOR_POS_MIN and DOOR_POS_MAX perhaps.